ta/arch/arm/ta.ld.S

```diff
@@ -8,14 +8,20 @@ OUTPUT_ARCH(aarch64)
 #endif
 
 PHDRS {
-	rodata PT_LOAD;
+	/*
+	 * Exec and rodata headers are hard coded to RX and RO
+	 * respectively. This is needed because the binary is relocatable
+	 * and the linker would automatically make any header writeable
+	 * that need to be updated during relocation.
+	 */
+	exec PT_LOAD FLAGS (5);		/* RX */
+	rodata PT_LOAD FLAGS (4);	/* RO */
 	rwdata PT_LOAD;
-	rodata2 PT_LOAD;
 	dyn PT_DYNAMIC;
 }
 
 SECTIONS {
-	.ta_head : {*(.ta_head)} :rodata
+	.ta_head : {*(.ta_head)} :exec
```

Comment:
It doesn't have to be exec, but adding it to a separate section wouldn't make a difference as it would get exec permissions any way because it's on the same page/section as some text.
At https://github.com/OP-TEE/optee_os/pull/740/files/a5131b04cf6a31315fe7ec4a7fe68ce77a364e4c#diff-5512ac6071f81d267fde55be2258ac82R349 
you can see that the permissions are or:ed together.